haproxy/haproxy-production.cfg

```diff
@@ -7,6 +7,7 @@ global
   user haproxy
   group haproxy
   daemon
+  maxconn 4096
```

Comment:
I was reading about what this does, but why did we not have it on our old configuration?

Reply:
@chiaberry wrote:

> I was reading about what this does, but why did we not have it on our old configuration?

Great question. I'm chalking it up to fundamental differences between the linux kernel on the centos machines (3.10.0) vs RHEL (5.14.0). RHEL's configuration isn't willing to give up a bajillion potential handles for connections, so we have to tell haproxy to settle down and "only" be ready for 4K incoming requests. 

Two major linux releases is about a an eon in linux development time too -- that's the biggest kernel version jump I've ever been a part of. Total Encino Man vibes.

Reply:
thanks for that explanation. 🙏 because i have no clue, did this surface through an error? trying to learn as i go 😅

Reply:
You bet! It did surface through an error. `haproxy` won't start at all if it isn't able to start with the configuration it has, so the container was croaking as soon as it fired up with an error that the kernel wasn't able to allocate enough handles -- by default, `haproxy` sets itself up to serve an enormous amount of traffic, like a `reddit.com` amount of traffic. `haproxy` will easily swamp a gigabit ethernet connection before it even comes close to bumping into a CPU restriction. It's just one of those "how do they do it" type of computing masterpieces.